src/extensions/typescript/typescript.compiler.ts

```diff
@@ -0,0 +1,54 @@
+import path from 'path';
+import ts from 'typescript';
+import { Compiler } from '../compile';
+
+export class TypescriptCompiler implements Compiler {
+  constructor(readonly tsConfig: Record<string, any>) {}
+
+  defineCompiler() {
+    return { taskFile: 'transpile' };
+  }
+  compileFile(
+    fileContent: string,
+    options: { componentDir: string; filePath: string }
+  ): { outputText: string; outputPath: string }[] | null {
+    const supportedExtensions = ['.ts', '.tsx'];
+    const fileExtension = path.extname(options.filePath);
+    if (!supportedExtensions.includes(fileExtension)) {
+      return null; // file is not supported
```

Comment:
change to undefined

Reply:
@GiladShoham why `undefined`? it is defined and it is null.